src/blocksplitter.rs

```diff
@@ -26,37 +26,39 @@ fn find_minimum<F>(f: F, start: usize, end: usize) -> (usize, f64)
     } else {
         /* Try to find minimum faster by recursively checking multiple points. */
         let num = 9;  /* Good value: 9. ?!?!?!?! */
-        let mut p = vec![0; num];
-        let mut vp = vec![0.0; num];
-        let mut besti;
+        let mut p;
+        let mut best_index;
         let mut best;
         let mut lastbest = f64::MAX;
-        let mut pos = start;
+        let mut pos      = start;
+        let mut range    = end - start;
 
-        while end - start > num {
-            for i in 0..num {
-                p[i] = start + (i + 1) * ((end - start) / (num + 1));
-                vp[i] = f(p[i]);
-            }
+        while range > num {
+            // Collect the values for `p`
+            p = (1..num+1).map(|i| start + i * (range / num)).collect::<Vec<usize>>();
 
-            besti = 0;
-            best = vp[0];
+            best_index = 0;
+            best = f(p[0]);
```

Comment:
These three variables all seem to be unconditionally set at the beginning of the loop; can we just move the declaration to here?